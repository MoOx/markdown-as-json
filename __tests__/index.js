// @flow

const markdownAsJsTree = require("../index.js").markdownAsJsTree;

it("should transform markdown as js", () => {
  expect(
    markdownAsJsTree(
      `---
test: a
test2: b
---
` + "## Test\n[link](href)\n```js\nconsole.log(window)\n```"
    )
  ).toMatchSnapshot();
});

it("should give front title", () => {
  expect(
    markdownAsJsTree(
      `---
title: Title front
---
` + "# Title md"
    )
  ).toMatchSnapshot();
});

it("should give fallback to md title", () => {
  expect(
    markdownAsJsTree(
      "# Title md"
    )
  ).toMatchSnapshot();
});

