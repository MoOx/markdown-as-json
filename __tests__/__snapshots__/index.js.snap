// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should transform markdown as json 1`] = `
Object {
  "body": Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "children": undefined,
                "props": Object {
                  "class": "icon icon-link",
                },
                "tag": "span",
              },
            ],
            "props": Object {
              "aria-hidden": "true",
              "href": "#test",
            },
            "tag": "a",
          },
          "Test",
        ],
        "props": Object {
          "id": "test",
        },
        "tag": "h2",
      },
      "
",
      Object {
        "children": Array [
          Object {
            "children": Array [
              "link",
            ],
            "props": Object {
              "href": "href",
            },
            "tag": "a",
          },
        ],
        "props": undefined,
        "tag": "p",
      },
      "
",
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "children": Array [
                  "console",
                ],
                "props": Object {
                  "class": "hljs-built_in",
                },
                "tag": "span",
              },
              ".log(",
              Object {
                "children": Array [
                  "window",
                ],
                "props": Object {
                  "class": "hljs-built_in",
                },
                "tag": "span",
              },
              ")",
            ],
            "props": Object {
              "class": "hljs language-js",
            },
            "tag": "code",
          },
        ],
        "props": undefined,
        "tag": "pre",
      },
    ],
    "props": undefined,
    "tag": "div",
  },
  "headings": Array [
    Object {
      "id": "test",
      "level": 2,
      "text": "Test",
    },
  ],
  "test": "a",
  "test2": "b",
  "title": undefined,
}
`;
